code/__HELPERS/global_lists.dm

```diff
@@ -57,15 +57,123 @@
+		/obj/item/stack/sheet/mineral/sandbags = GLOB.sandbag_recipes,
+		/obj/item/stack/sheet/mineral/diamond = GLOB.diamond_recipes,
+		/obj/item/stack/sheet/mineral/uranium = GLOB.uranium_recipes,
+		/obj/item/stack/sheet/mineral/plasma = GLOB.plasma_recipes,
+		/obj/item/stack/sheet/mineral/gold = GLOB.gold_recipes,
+		/obj/item/stack/sheet/mineral/silver = GLOB.silver_recipes,
+		/obj/item/stack/sheet/mineral/bananium = GLOB.bananium_recipes,
+		/obj/item/stack/sheet/mineral/titanium = GLOB.titanium_recipes,
+		/obj/item/stack/sheet/mineral/plastitanium = GLOB.plastitanium_recipes,
+		/obj/item/stack/sheet/mineral/snow = GLOB.snow_recipes,
+		/obj/item/stack/sheet/mineral/adamantine = GLOB.adamantine_recipes,
+		/obj/item/stack/sheet/mineral/abductor = GLOB.abductor_recipes,
+		/obj/item/stack/sheet/iron = GLOB.metal_recipes,
+		/obj/item/stack/sheet/plasteel = GLOB.plasteel_recipes,
+		/obj/item/stack/sheet/mineral/wood = GLOB.wood_recipes,
+		/obj/item/stack/sheet/mineral/bamboo = GLOB.bamboo_recipes,
+		/obj/item/stack/sheet/cloth = GLOB.cloth_recipes,
+		/obj/item/stack/sheet/durathread = GLOB.durathread_recipes,
+		/obj/item/stack/sheet/cardboard = GLOB.cardboard_recipes,
+		/obj/item/stack/sheet/bronze = GLOB.bronze_recipes,
+		/obj/item/stack/sheet/plastic = GLOB.plastic_recipes,
+		/obj/item/stack/ore/glass = GLOB.sand_recipes,
+		/obj/item/stack/rods = GLOB.rod_recipes,
+		/obj/item/stack/sheet/runed_metal = GLOB.runed_metal_recipes,
+	)
+
+	for(var/stack in global_stack_recipes)
+		for(var/stack_recipe in global_stack_recipes[stack])
+			if(istype(stack_recipe, /datum/stack_recipe_list))
+				var/datum/stack_recipe_list/stack_recipe_list = stack_recipe
+				for(var/nested_recipe in stack_recipe_list.recipes)
+					var/datum/crafting_recipe/stack/recipe = new/datum/crafting_recipe/stack(stack, nested_recipe)
+					if(recipe.name != "" && recipe.result)
+						GLOB.crafting_recipes += recipe
+			else
+				var/datum/crafting_recipe/stack/recipe = new/datum/crafting_recipe/stack(stack, stack_recipe)
+				if(recipe.name != "" && recipe.result)
+					GLOB.crafting_recipes += recipe
+
+	var/list/material_stack_recipes = list(
+		SSmaterials.base_stack_recipes,
+		SSmaterials.rigid_stack_recipes,
+	)
+
+	for(var/list/recipe_list in material_stack_recipes)
+		for(var/stack_recipe in recipe_list)
+			var/datum/crafting_recipe/stack/recipe = new/datum/crafting_recipe/stack(/obj/item/stack/sheet/iron, stack_recipe)
+			recipe.steps = list("Use different materials in hand to make an item of that material")
+			GLOB.crafting_recipes += recipe
+
+/// Inits atoms used in crafting recipes
+/proc/init_crafting_recipes_atoms()
+	var/list/recipe_lists = list(
+		GLOB.crafting_recipes,
+		GLOB.cooking_recipes
+	)
+	var/list/atom_lists = list(
+		GLOB.crafting_recipes_atoms,
+		GLOB.cooking_recipes_atoms
+	)
+
+	for(var/recipe_list in recipe_lists)
+		for(var/R in recipe_list)
+			var/datum/crafting_recipe/recipe = R
```

Comment:
```suggestion
		for(var/datum/crafting_recipe/recipe as anything in recipe_list)
```

Reply:
Oh I didn't know you can typecast like this. I will change all such places then.